### PR TITLE
Run pipeline aggregations for empty buckets added in the Range Aggregation

### LINKED ISSUE #15471: Bucket selector script is not applied to empty histogram buckets
Consider this dataset:

```
POST _bulk
{"index": { "_index": "aggtest", "_type": "default" }}
{"a": 1, "b": 1}
{"index": { "_index": "aggtest", "_type": "default" }}
{"a": 1, "b": 2}
{"index": { "_index": "aggtest", "_type": "default" }}
{"a": 3, "b": 1}
{"index": { "_index": "aggtest", "_type": "default" }}
{"a": 3, "b": 3}
```

Run this aggregation on it:
```
POST aggtest/default/_search?size=0
{
  "aggs": {
    "by_a": {
      "histogram": {
        "field": "a",
        "interval": 1
      },
      "aggs": {
        "by_b": {
          "range": {
            "field": "b",
            "ranges": [
              { "from": 1, "to": 2 },
              { "from": 2, "to": 3 },
              { "from": 3, "to": 4 }
            ]
          },
          "aggs": {
            "the_filter": {
              "bucket_selector": {
                "buckets_path": {
                  "the_doc_count": "_count"
                },
                "script": {
                  "inline": "the_doc_count > 0",
                  "lang": "expression"
                }
              }
            }
          }
        }
      }
    }
  }
}
```

This is the result. Note how the empty buckets of `by_b` are not removed by the script when the parent bucket of `by_a` is empty.

Result

### PR DESCRIPTION
**Pull request by colings86**
Closes #15471

### CODE REVIEW

File: modules/lang-groovy/src/test/java/org/elasticsearch/messy/tests/BucketSelectorTests.java [hunk 1]
```diff
@@ -30,28 +49,11 @@
...
-
-import static org.elasticsearch.common.xcontent.XContentFactory.jsonBuilder;
-import static org.elasticsearch.search.aggregations.AggregationBuilders.histogram;
-import static org.elasticsearch.search.aggregations.AggregationBuilders.sum;
-import static org.elasticsearch.search.aggregations.pipeline.PipelineAggregatorBuilders.having;
-import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertSearchResponse;
-import static org.hamcrest.Matchers.equalTo;
-import static org.hamcrest.Matchers.greaterThan;
-import static org.hamcrest.Matchers.lessThan;
-import static org.hamcrest.Matchers.notNullValue;
```
Comment (jpountz) [inline modules/lang-groovy/src/test/java/org/elasticsearch/messy/tests/BucketSelectorTests.java]: hmm, why did imports move that much?

File: modules/lang-groovy/src/test/java/org/elasticsearch/messy/tests/BucketSelectorTests.java [hunk 2]
```diff
@@ -451,4 +463,47 @@ public void testPartiallyUnmapped() throws Exception {
             assertThat(field2SumValue + field3SumValue, greaterThan(100.0));
         }
     }
+
+    public void testEmptyBuckets() {
+        SearchResponse response = client().prepareSearch("idx_with_gaps")
+                .addAggregation(histogram("histo").field(FIELD_1_NAME).interval(1)
+                        .subAggregation(range("range").field(FIELD_2_NAME).addRange(1, 2).addRange(2, 3).addRange(3, 4)
+                                .subAggregation(having("bucket_selector").setBucketsPaths("_count").script(new Script("_value0 > 0")))))
```
Comment (jpountz) [inline modules/lang-groovy/src/test/java/org/elasticsearch/messy/tests/BucketSelectorTests.java]: is it possible to write this test in a way that it either does not require a script, or uses a mock script?

**Comment by jpountz** [comment]
The fix looks good to me, it's surprisingly simple! I'm more concerned about the test however: can we make it not use an actual script?

File: modules/lang-groovy/src/test/java/org/elasticsearch/messy/tests/BucketSelectorTests.java (hunk 1 above)
Comment (colings86) [inline modules/lang-groovy/src/test/java/org/elasticsearch/messy/tests/BucketSelectorTests.java]: I think something might have changed in the import ordering in the IDE. Not sure if that was a project setting changed in master or if it's just my Eclipse, I will check.

**Comment by colings86** [comment]
@jpountz I pushed a new commit which changes the test so it doesn't use scripts

**Comment by jpountz** [comment]
LGTM

**Pull request merged**